projects/components/src/datagrid/datagrid.component.ts

```diff
@@ -520,6 +535,47 @@ export class DatagridComponent<R> implements OnInit, AfterViewInit {
 
     private viewInitted = false;
 
+    /**
+     * To add or replace a column of this datagrid columns. Exposed for columns modifiers(eg: directives) that listen to
+     * {@link columnsUpdated} event and want to modify the columns set by components using this datagrid.
+     */
+    public addColumn(col: GridColumn<R>): void {
+        if (!col) {
+            return;
+        }
+        const colIndex = this.findColumnIndex(col);
+        if (colIndex !== null) {
+            this._columns[colIndex] = col;
+        } else {
+            this._columns.push(col);
+        }
+        this.updateColumnsConfig();
+    }
+
+    /**
+     * To remove a column from this datagrid columns. Exposed for columns modifiers(eg: directives) that listen to
+     * {@link columnsUpdated} event and want to modify the columns set by components using this datagrid.
+     */
+    public removeColumn(col: GridColumn<R>): void {
+        if (!col) {
+            return;
+        }
+        const colIndex = this.findColumnIndex(col);
+        if (colIndex !== null) {
+            this._columns.splice(colIndex, 1);
+            this.updateColumnsConfig();
+        }
+    }
+
+    private findColumnIndex(col: GridColumn<R>): number {
```

Comment:
Why return null instead of -1? Most `findIndex` methods return -1 when an index is not found